lib/duo.c

```diff
@@ -324,6 +327,49 @@ _duo_bson_response(struct duo_ctx *ctx, bson *resp)
     return (ret);
 }
 
+static int
+_sleep_ms(long milliseconds)
+{
+    struct timespec tim;
+    tim.tv_sec = milliseconds / 1000;
+    tim.tv_nsec = (milliseconds % 1000) * 1000000;
+
+    if (nanosleep(&tim, &tim) != 0) {
+        /* keep resuming nanosleep until it's no longer interrupted by a signal */
+        while (errno == EINTR) {
+            if (nanosleep(&tim, &tim) == 0) {
+                break;
+            }
+        }
+        return -1;
```

Comment:
Hey Marcus, can you explain why we wanna return `-1` here? Because of your while loop it seems like we could successfully get through the sleep, but still end up returning `-1`. Am I understanding correctly?

Reply:
That's a bug, good catch. I'll fix it if it turns out that nanosleep is the right thing to use here.